tests/File/CommonFilePermissionsTest.php

```diff
@@ -0,0 +1,136 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Keboola\Test\File;
+
+use Keboola\StorageApi\ClientException;
+use Keboola\StorageApi\Options\FileUploadOptions;
+use Keboola\StorageApi\Options\ListFilesOptions;
+use Keboola\StorageApi\Options\TokenCreateOptions;
+use Keboola\StorageApi\Options\TokenUpdateOptions;
+use Keboola\Test\StorageApiTestCase;
+
+class CommonFilePermissionsTest extends StorageApiTestCase
+{
+    public function testFilesPermissions(): void
+    {
+        $filePath = __DIR__ . '/../_data/files.upload.txt';
+        $uploadOptions = new FileUploadOptions();
+
+        $tokenOptions = (new TokenCreateOptions())
+            ->setDescription('Files test')
+        ;
+
+        $newToken = $this->tokens->createToken($tokenOptions);
+
+        $this->createAndWaitForFile($filePath, $uploadOptions);
+
+        $totalFilesCount = count($this->_client->listFiles());
+        $this->assertNotEmpty($totalFilesCount);
+
+        // new token should not have access to any files
+        $newTokenClient = $this->getClient([
+            'token' => $newToken['token'],
+            'url' => STORAGE_API_URL,
+        ]);
+
+        $this->assertEmpty($newTokenClient->listFiles());
+
+        $newFileId = $this->createAndWaitForFile($filePath, $uploadOptions, $newTokenClient);
+        /** @var array<array{id:int}> $files */
+        $files = $newTokenClient->listFiles();
+        $this->assertCount(1, $files);
+        /** @var array{id:int} $reset */
+        $reset = reset($files);
+        $this->assertEquals($newFileId, $reset['id']);
+
+        // new file should be visible for master token
+        $files = $this->_client->listFiles();
+        /** @var array{id:int} $reset */
+        $reset = reset($files);
+        $this->assertEquals($newFileId, $reset['id']);
+
+        $this->tokens->dropToken($newToken['id']);
+    }
+
+    public function testFilesPermissionsCanReadAllFiles(): void
+    {
+        $uploadOptions = new FileUploadOptions();
+        $uploadOptions->setFileName('test.txt')
+            ->setFederationToken(true);
+        $file = $this->_client->prepareFileUpload($uploadOptions);
+
+        $tokenOptions = (new TokenCreateOptions())
+            ->setDescription('Files test')
+            ->setCanReadAllFileUploads(true)
+        ;
+
+        $newToken = $this->tokens->createToken($tokenOptions);
+
+        // new token should not have access to any files
```

Comment:
to je asi copy paste ne?

znamena to, ze novy token nemuze listovat fily (viz predchozi test) ale, muze je cist, pokud vi ID?

Reply:
pak bych tam presunul tento komentar https://github.com/keboola/storage-api-php-client/pull/1143#discussion_r1276173816

Reply:
jo tie testy su vsetky C&P nechapem totok `pak bych tam presunul tento komentar https://github.com/keboola/storage-api-php-client/pull/1143#discussion_r1276173816` to je uplne nieco ine

Reply:
jo, tak jsem to myslel. V tom linku jsem chtel zdokumentovat co dela ten merge provideru. Tak jsi na to udelal metodu a ten komentar jsi tam dal. takze cajk.